File: src/main/java/br/com/jiratorio/service/LeadTimeConfigService.java
package br.com.jiratorio.service;

import br.com.jiratorio.domain.entity.LeadTimeConfig;
import br.com.jiratorio.domain.request.LeadTimeConfigRequest;
import br.com.jiratorio.domain.response.LeadTimeConfigResponse;
import java.util.List;

public interface LeadTimeConfigService {

    List<LeadTimeConfig> findAllByBoardId(Long boardId);

    List<LeadTimeConfigResponse> findAll(Long boardId);

    Long create(Long boardId, LeadTimeConfigRequest leadTimeConfigRequest);

    LeadTimeConfigResponse findByBoardAndId(Long boardId, Long id);

    void update(Long boardId, Long id, LeadTimeConfigRequest leadTimeConfigRequest);

    void deleteByBoardAndId(Long boardId, Long id);
}

File: src/main/java/br/com/jiratorio/service/LeadTimeConfigService.kt
package br.com.jiratorio.service

import br.com.jiratorio.domain.entity.LeadTimeConfig
import br.com.jiratorio.domain.request.LeadTimeConfigRequest
import br.com.jiratorio.domain.response.LeadTimeConfigResponse

interface LeadTimeConfigService {

    fun findAllByBoardId(boardId: Long): List<LeadTimeConfig>

    fun findAll(boardId: Long): List<LeadTimeConfigResponse>

    fun create(boardId: Long, leadTimeConfigRequest: LeadTimeConfigRequest): Long

    fun findByBoardAndId(boardId: Long, id: Long): LeadTimeConfigResponse

    fun update(boardId: Long, id: Long, leadTimeConfigRequest: LeadTimeConfigRequest)

    fun deleteByBoardAndId(boardId: Long, id: Long)
}
